File: from-0201-to-0300/0283-move-zeroes/solution.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        p1, p2 = 0, 0
        while p2 < len(nums):
            if nums[p2] != 0:
                nums[p1] = nums[p2]
                p1 += 1
            p2 += 1

        while p1 < len(nums):
            nums[p1] = 0
            p1 += 1

File: from-0201-to-0300/0283-move-zeroes/solution.go
func moveZeroes(nums []int)  {
    ps, pf := 0, 0
    for ps < len(nums) && nums[ps] != 0 {
        ps += 1
    }
    pf = ps + 1
    for pf < len(nums) {
        if nums[pf] != 0 {
            nums[ps] = nums[pf]
            ps += 1
        }
        pf += 1
    }
    for ps < len(nums) {
        nums[ps] = 0
        ps += 1
    }
}
